confidential-data-hub/hub/src/bin/config/mod.rs

```diff
@@ -65,13 +75,31 @@ impl Default for CdhConfig {
 }
 
 impl CdhConfig {
-    pub async fn init(config_path: &str) -> Result<Self> {
-        let mut config = Self::from_file(config_path).unwrap_or_else(|e| {
-            warn!("read config file {config_path} failed {e:?}, use a default config where `aa_kbc_params` = offline_fs_kbc::null.");
-            Self::default()
+    pub fn new(config_path: Option<String>) -> Result<Self> {
+        let config_path = config_path.or_else(|| {
+            if let std::result::Result::Ok(env_path) = env::var("CDH_CONFIG_PATH") {
+                debug!("Read CDH's config path from env: {env_path}");
+                return Some(env_path);
+            }
+            None
         });
 
-        config.update_from_kernel_cmdline().await?;
+        let mut config = match config_path {
+            Some(path) => {
+                info!("Use configuration file {path}");
+                if !Path::new(&path).exists() {
+                    bail!("Config file {path} not found.")
+                }
+
+                Self::from_file(&path)?
+            }
+            None => {
+                info!("No config path specified, use a default one.");
+                Self::default()
+            }
+        };
+
+        config.update_from_kernel_cmdline()?;
```

Comment:
should we rename that to `extend_credentials_from_kernel_cmdline()` and/or add a comment what this fn does? I was initially confused why the kernel cmdline can override the config again, but it seems to only extend the credentials vector.

Reply:
Nice idea. Let me rename the fn